docs: rego rule explanation and links (#519)
stackabletech/hdfs-operator: docs/modules/hdfs/pages/usage-guide/security.adoc

----

This rego rule is intended for demonstration purposes and allows every operation.
For a production setup you will probably need to have something much more granular.

> We provide a more representative rego rule in our integration tests and in the afore-mentioned hdfs-utils repository.

Reviewer: Had to google this word, IDK if this is correct
```suggestion
We provide a more representative rego rule in our integration tests and in the aforementioned hdfs-utils repository.
```

> Details can be found below in the <<_fine_granular_rego_rule_, fine-granular rego rule>> section.

Reviewer 2: ```suggestion
Details can be found below in the <<fine-granular-rego-rules, fine-granular rego rules>> section.
```

We have decided that this is an acceptable approach as normal operations will not be affected.
In case you really need users to be part of the `superusers` group, you can use a configOverride on `hadoop.user.group.static.mapping.overrides` for that.

> [#_fine_granular_rego_rule_]
> === Fine-granular rego rule

Reviewer 2: ```suggestion
[#fine-granular-rego-rules]
=== Fine-granular rego rules
```

The hdfs-utils repository contains a more production-ready rego-rule https://github.com/stackabletech/hdfs-utils/blob/main/rego/hdfs.rego[here].

> With a few minor differences (e.g. pod names) it is the same rego rule that is used in this https://github.com/stackabletech/hdfs-operator/blob/main/tests/templates/kuttl/kerberos/12-rego-rules.txt.j2[integration test].

Reviewer 2: ```suggestion
With a few minor differences (e.g. Pod names) it is the same rego rule that is used in this https://github.com/stackabletech/hdfs-operator/blob/main/tests/templates/kuttl/kerberos/12-rego-rules.txt.j2[integration test].
```
